app/template_extractor.py

```diff
@@ -0,0 +1,314 @@
+# TODO
+"""
+- Get unit tests set up for running of script (e.g. against different JSON, outputs etc)
+
+- Get values from dict recursively instead of multiple nested loops?
+
+- Think of better way to handle intermittent keys in get_conditional_text_for_container().
+
+- Add/amend other team/best practice stuff that's been missed out.
+"""
+import argparse
+import datetime
+import os
+from collections import defaultdict
+from app.utils import get_output_file_path, deserialise_json
+
+
+class TemplateExtractor:
+
+    def __init__(self, input_schema_path, output_directory, output_format='pot'):
+        self.input_schema_path = input_schema_path
+        self.schema_data = deserialise_json(input_schema_path)
+        self.output_format = output_format
+        self.output_path = get_output_file_path(input_schema_path, output_directory, output_format)
+
+        if not os.path.exists(os.path.dirname(self.output_path)):
+            os.makedirs(os.path.dirname(self.output_path))
+
+        self.keys_for_translation = ['description', 'label', 'title', 'question', 'legal_basis']
+        self.translation_object = defaultdict()
+        self._create_translation_object()
+
+    def create_template(self):
+        print('\nCreating {} file {} from {}'.format(self.output_format, self.output_path, self.input_schema_path))
+
+        if self.output_format == 'csv':
+            self.write_to_csv_file()
+        else:
+            self.write_to_pot_file()
+
+    def _create_translation_object(self):
+        if self.schema_data.get('legal_basis') != 'Voluntary':
```

Comment:
Ran it directly in PyCharm. 
Getting `KeyError` on `legal_basis` so this could be something like this?
```py
	legal_basis = self.schema_data.get('legal_basis')
	if legal_basis and legal_basis != 'Voluntary':
```

Reply:
Most of my work on this has been refactoring existing methods to be better named - which census schema are you trying to translate? The v3 branch on census-schemas works for me.

Reply:
This PR was originally about the functional behaviour, so that should definitely be included here.